admin/class-cornerstone.php

```diff
@@ -23,14 +29,57 @@ public function register_hooks() {
 	 * @param int $post_id The post id to save the meta value for.
 	 */
 	public function save_meta_value( $post_id ) {
-		$is_cornerstone = ( filter_input( INPUT_POST, self::META_NAME ) === '1' );
+		$is_cornerstone = $this->is_cornerstone_checkbox();
 
 		if ( $is_cornerstone ) {
-			update_post_meta( $post_id, self::META_NAME, $is_cornerstone );
+			$this->update_meta( $post_id, $is_cornerstone );
 
 			return;
 		}
 
+		$this->delete_meta( $post_id );
+	}
+
+	/**
+	 * Returns the result of the cornerstone checkbox.
+	 *
+	 * @return bool True when checkbox is checked.
+	 */
+	protected function is_cornerstone_checkbox() {
+		return filter_input( INPUT_POST, self::META_NAME ) === '1';
+	}
+
+	/**
+	 * Checks if the current page matches one of the pages that contains the cornerstone content field.
+	 *
+	 * @param string $page The page to check.
+	 *
+	 * @return bool True when the page contains the cornerstone field.
+	 */
+	protected function page_contains_cornerstone_field( $page ) {
```

Comment:
Consider keeping the name aligned with the comment (i.e. `page_contains_cornerstone_content_field`)